packages/salesforce-adapter/src/transformers/reference_mapping.ts

```diff
@@ -1764,7 +1790,18 @@ const getLookUpNameImpl = ({
       log.debug('could not determine field for path %s', args.path?.getFullName())
       return undefined
     }
-    const strategies = (await resolverFinder(args.field, args.element)).map(def => def.serializationStrategy)
+    const definitions = await resolverFinder(args.field, args.element)
+    const filteredDefinitions = definitions.filter(def => {
+      if (def.target === undefined || def.target.type === undefined || !isElement(args.ref.value)) {
+        log.debug('could not filter serialization strategies by target type for field %o', def)
```

Comment:
```suggestion
        log.debug('could not filter serialization strategy from def %s', inspectValue(def))
```
Also might remove this log, as I think it can be very noisy. (Or set as trace)